server/routes.js

```diff
@@ -0,0 +1,166 @@
+var db = require('./db')
+
+module.exports = {
+  getDonors: getDonors,
+  getRecipients: getRecipients,
+  getDonor: getDonor,
+  getRecipient: getRecipient,
+  updateTicket: updateTicket,
+  updateComment: updateComment,
+  getTickets: getTickets,
+  addTicket: addTicket,
+  getDonorTicket: getDonorTicket,
+  getRecipientTicket: getRecipientTicket,
+  getTicketComments: getTicketComments
+}
+
+function getDonors(req, res) {
+  db.getDonors()
+    .then(function (donors) {
+      res.json(donors)
+    })
+    .catch(function (err) {
+      res.send(err.message).status(500)
+    })
+}
+
+function getRecipients(req, res) {
+  db.getRecipients()
+    .then(function (recipients) {
+      res.json(recipients)
+    })
+    .catch(function (err) {
+      res.send(err.message).status(500)
+    })
+}
+
+function getDonor(req, res) {
+  var donorId = req.params.id
+  db.getDonor(donorId)
+    .then(function (donor) {
+      res.json(donor)
+    })
+    .catch(function (err) {
+      res.send(err.message).status(500)
+    })
+}
+
+function getRecipient(req, res) {
+  var recipientId = req.params.id
+  db.getRecipient(recipientId)
+    .then(function (recipient) {
+      res.json(recipient)
+    })
+    .catch(function (err) {
+      res.send(err.message).status(500)
+    })
+}
+
+function updateTicket(req, res) {
+  var ticket = {
+    id: req.body.ticketId,
+    actualKg: req.body.actualKg,
+  }
+  db.updateTicket(ticket)
+    .then(function () {
+      res.json(ticket)
+    })
+    .catch(function (err) {
+      res.send(err.message).status(500)
+    })
+}
+
+function updateComment(req, res) {
+  var comment = {
+    ticketId: req.body.ticketId,
+    comment: req.body.comment
+  }
+  db.updateComment(comment)
+    .then(function () {
+      res.json(comment)
+    })
+    .catch(function (err) {
+      res.send(err.message).status(500)
+    })
+}
+
+// "has to match table column name, use _": req.body.useCamelCase
+  function addTicket(req, res) {
+    console.log(req.body.recipientId)
+    if (!req.body.recipientId) {
+      var ticket = {
+        expected_kg: req.body.expectedKg,
+        donor_id: req.body.donorId.split('|')[0],
+        details_id: req.body.donorId.split('|')[1],
+        is_complete: 0
+      }
+    } else {
+      var ticket = {
+        expected_kg: req.body.expectedKg,
+        recipient_id: req.body.recipientId.split('|')[0],
+        details_id: req.body.recipientId.split('|')[1],
+        is_complete: 0
+      }
+    }
+    console.log(ticket)
+    db.addTicket(ticket)
+      .then(function () {
+        res.json(ticket)
```

Comment:
This will return an HTTP status code of 200 by default. You want 201 CREATED for a POST request.